### PR TITLE
feat: implement OpenAPI server URL template variable support

### LINKED ISSUE #23: Feature: Add support for OpenAPI server URL template variables
## Summary
Add support for OpenAPI server URL template variables through CLI flags, allowing users to dynamically specify server parameters at runtime.

## Background
PR #22 fixed a panic when encountering server URL templates by detecting them and providing an error message. However, this requires users to manually configure base URLs using `aperture config set-url`. 

OpenAPI 3.0 supports server URL templates like:
```yaml
servers:
  - url: https://{region}.sentry.io
    variables:
      region:
        default: us
        enum: [us, eu]
        description: The regional instance of Sentry
```

## Proposed Solution
Add CLI support for server variables, allowing users to specify them at runtime:
```bash
# Using a flag
aperture api sentry events list-events --server-var region=eu --organization my-org --project my-project

# Or with multiple variables
aperture api multi-region health ping --server-var region=us --server-var env=staging
```

## Implementation Details
1. Add a new `--server-var` flag that accepts key=value pairs
2. Parse and validate server variables against the OpenAPI spec's variable definitions
3. Substitute variables in the server URL before making requests
4. Honor default values from the OpenAPI spec when variables aren't provided
5. Validate against enum constraints if specified in the OpenAPI spec

## Benefits
- Eliminates the need for manual `aperture config set-url` for APIs with template variables
- Supports dynamic server selection at runtime
- Aligns with OpenAPI 3.0 specification capabilities
- Improves developer experience for multi-region/multi-environment APIs

## Related
- PR #22: Initial fix to prevent panic with server URL templates
- OpenAPI 3.0 Server Object specification: https://spec.openapis.org/oas/v3.0.3.html#server-object

### PR DESCRIPTION
**Pull request by kioku**
## Summary

Implements comprehensive support for OpenAPI 3.0 server URL template variables, enabling dynamic server selection for multi-region and multi-environment APIs.

### Key Features

- **Dynamic Server Variables**: Support for `{variable}` patterns in OpenAPI server URLs  
- **CLI Integration**: New `--server-var key=value` global flag for runtime variable specification
- **Validation & Constraints**: Full enum constraint validation and required variable checking
- **Intelligent Defaults**: Automatic fallback to OpenAPI spec default values
- **Enhanced Error Handling**: Early detection of unresolvable templates with clear error messages
- **Backward Compatibility**: Maintained through fallback mechanisms and consistent API behavior

### Implementation Details

**Data Models**: Added `ServerVariable` struct and extended `CachedSpec` with server variable support  
**OpenAPI Integration**: Extract server variables from OpenAPI 3.x specifications during transformation  
**CLI Interface**: Global `--server-var` flag with multi-value support and proper argument parsing  
**Resolution Engine**: Comprehensive validation, enum checking, and URL template substitution  
**URL Resolution**: Enhanced BaseUrlResolver with early template validation and improved error handling  
**Request Execution**: Integrated server variable resolution in the complete request lifecycle

### Usage Examples

```bash
# Multi-region API calls with explicit variables
aperture api sentry issues list --server-var region=eu --server-var env=prod

# Using OpenAPI spec defaults for optional variables
aperture api stripe customers get --id cus_123 --server-var env=staging

# Validation errors provide clear guidance
aperture api my-api users get --id 123 --server-var region=invalid
# Error: Invalid value 'invalid' for server variable 'region'. Allowed values: us, eu, ap

# Dry-run to inspect resolved URLs
aperture --dry-run api regional-api health get-status --server-var env=staging
# GET https://staging.api.example.com/status
```

### Technical Changes

- **Cache Format**: Incremented to version 3 for server variable support
- **Error Types**: Added 6 new error variants (`UnresolvedTemplateVariable`, `MissingServerVariable`, etc.) with structured JSON output
- **URL Resolution**: Enhanced `BaseUrlResolver` to detect template issues early and provide better error context
- **Test Coverage**: 275+ tests pass including 19 dedicated server variable tests and comprehensive integration tests
- **Code Quality**: All clippy lints pass with pedantic settings, comprehensive documentation added

### Error Handling Improvements

- **Early Template Detection**: URLs with unresolvable template variables fail during URL resolution rather than path parameter substitution
- **Contextual Error Messages**: `UnresolvedTemplateVariable` errors include both variable name and full URL context
- **Graceful Fallbacks**: The `resolve()` method maintains backward compatibility by falling back to basic resolution when server variable resolution fails
- **Comprehensive Validation**: Server variable format, enum constraints, and required variable validation with actionable error messages

### Breaking Changes

None. All changes maintain backward compatibility:
- Existing APIs without server variables continue to work unchanged
- Non-template URLs pass through without modification  
- Legacy URL resolution paths are preserved through fallback mechanisms
- Error types changed from `MissingPathParameter` to `UnresolvedTemplateVariable` for better error context, but this only affects template URLs with undefined server variables (an error condition in both cases)

## Test Results

**All tests passing**: 275+ tests across 35 test suites  
**Server Variable Tests**: 19 comprehensive tests covering validation, defaults, substitution, and error cases  
**Integration Tests**: End-to-end testing with `--server-var` flag, dry-run validation, and error scenarios  
**Backward Compatibility**: Verified through dedicated tests for non-template URLs and undefined server variable scenarios  
**Code Quality**: All clippy lints pass with strict settings (`-D warnings`)

### Key Test Categories
- Server variable resolution with all provided values
- Default value fallback behavior  
- Required variable validation
- Enum constraint enforcement
- Template substitution edge cases
- Backward compatibility for legacy URLs
- Error handling and message clarity
- CLI flag parsing and validation

Resolves #23

### CODE REVIEW

File: src/spec/transformer.rs [hunk 1]
```diff
@@ -141,6 +141,34 @@ impl SpecTransformer {
...
+                                            Some(variable.default.clone())
+                                        },
+                                        enum_values: variable.enumeration.clone(),
+                                        description: variable.description.clone(),
+                                    },
+                                )
+                            })
+                            .collect()
+                    })
+            });
```
Comment (kioku) [inline src/spec/transformer.rs:170]: simplify the nested map_or_else into a single iterator chain

File: src/config/url_resolver.rs [hunk 2]
```diff
@@ -43,8 +45,91 @@ impl<'a> BaseUrlResolver<'a> {
...
+
+            if !template_vars.is_empty() {
+                return Err(Error::UnresolvedTemplateVariable {
+                    name: template_vars[0].clone(),
+                    url: base_url,
+                });
+            }
+
+            return Ok(base_url);
+        }
```
Comment (kioku) [inline src/config/url_resolver.rs:123]: refactor this section for readability

File: src/config/server_variable_resolver.rs [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+                } else {
+                    return Err(Error::UnresolvedTemplateVariable {
+                        name: var_name.to_string(),
+                        url: url_template.to_string(),
+                    });
+                }
+            } else {
+                break;
+            }
+        }
```
Comment (kioku) [inline src/config/server_variable_resolver.rs:120]: refactor for readability

File: src/config/server_variable_resolver.rs [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+                });
+            }
+
+            Ok((key.to_string(), value.to_string()))
+        } else {
+            Err(Error::InvalidServerVarFormat {
+                arg: arg.to_string(),
+                reason: "Expected format: key=value".to_string(),
+            })
+        }
```
Comment (kioku) [inline src/config/server_variable_resolver.rs:151]: refactor for readability by using an early return

File: src/config/server_variable_resolver.rs [hunk 5]
```diff
@@ -0,0 +1,475 @@
...
+        let mut chars = name.chars();
+        if let Some(first_char) = chars.next() {
+            if !first_char.is_ascii_alphabetic() && first_char != '_' {
+                return Err(Error::InvalidServerVarFormat {
+                    arg: format!("{{{name}}}"),
+                    reason: "Template variable names must start with a letter or underscore"
+                        .to_string(),
+                });
+            }
+        }
```
Comment (kioku) [inline src/config/server_variable_resolver.rs:197]: flatten the nested if statements

**Review by kioku — COMMENTED** [review]
Handle the inline comments.
